cmd/tailscale/cli/ssh_unix.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !js && !windows
+// +build !js,!windows
+
+package cli
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"golang.org/x/sys/unix"
+)
+
+func init() {
+	getSSHClientEnvVar = func() string {
+		if os.Getenv("SUDO_USER") == "" {
+			// No sudo, just check the env.
+			return os.Getenv("SSH_CLIENT")
+		}
+		// SID is the session ID of the user's login session.
+		// It is also the process ID of the original shell that the user logged in with.
+		// We only need to check the environment of that process.
+		sid, err := unix.Getsid(os.Getpid())
+		if err != nil {
+			return ""
+		}
+		b, err := os.ReadFile(filepath.Join("/proc", strconv.Itoa(sid), "environ"))
```

Comment:
This is a Linuxism but build tags are for Unix

Reply:
There doesn't seem to be a decent way to detect this on darwin, leaving a TODO to fix later.

Reply:
`ps -E` doesn't provide any information unfortunately

Reply:
`ps eww $PID` works

Reply:
Nope
```
➜  ~ ssh maisem@localhost                                                                                                                                                                                          ~
(maisem@localhost) Password:
Last login: Tue May 31 10:45:57 2022 from ::1
➜  ~ ps eww  $$                                                                                                                                                                                                    ~
  PID   TT  STAT      TIME COMMAND
 4637 s007  Ss     0:00.11 -zsh
➜  ~
```